File: src/main/java/com/ticket/demo/model/Ticket.java
package com.ticket.demo.model;

import java.util.Date;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.ForeignKey;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

@Entity 
@Table (name="T_TICKET")
public class Ticket {
	
	@Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
	
	private String title;
	
	private Date startDate;
	
	private Date endDate;
	
	private String position; 
	
	private Double price;
	
	private Integer quantite;
	
	private Integer limitAge;
	
	@ManyToOne(cascade = CascadeType.ALL)
	@JoinColumn(name="category_id", referencedColumnName = "id", foreignKey = @ForeignKey(name="fk_ticket_category"))
	private Category category;
	
	public Ticket() {
	
	}

	public void setId(Integer id) {
		this.id = id;
	}

	public Integer getId() {
		return id;
	}

	public String getTitle() {
		return title;
	}

	public void setTitle(String title) {
		this.title = title;
	}

	public Date getStartDate() {
		return startDate;
	}

	public void setStartDate(Date startDate) {
		this.startDate = startDate;
	}

	public Date getEndDate() {
		return endDate;
	}

	public void setEndDate(Date endDate) {
		this.endDate = endDate;
	}

	public String getPosition() {
		return position;
	}

	public void setPosition(String position) {
		this.position = position;
	}

	public Double getPrice() {
		return price;
	}

	public void setPrice(Double price) {
		this.price = price;
	}

	public Integer getQuantite() {
		return quantite;
	}

	public void setQuantite(Integer quantite) {
		this.quantite = quantite;
	}

	public Integer getLimitAge() {
		return limitAge;
	}

	public void setLimitAge(Integer limiAge) {
		this.limitAge = limiAge;
	}

	public Category getCategory() {
		return category;
	}

	public void setCategory(Category category) {
		this.category = category;
	}

}

File: src/app/utils/models/Ticket.ts
import {Category} from "./Category";

export class Ticket
{
  public id: number = 0;

  public title: string = '';

  public price: number | undefined;

  public quantite: number = 0;

  public startDate : Date | undefined;

  public endDate: Date | undefined;

  public position: string | undefined;

  public limitAge: number = 1;

  public category: Category;

  constructor(newTicket: Ticket, desc: string) {
    this.id = newTicket.id;
    this.title = newTicket.title;
    this.price = newTicket.price;
    this.quantite = newTicket.quantite;
    this.startDate = newTicket.startDate;
    this.endDate = newTicket.endDate;
    this.position = newTicket.position;
    this.limitAge = newTicket.limitAge;
    this.category = new Category(desc);
  }
}
